packages/relay-service/src/index.js

```diff
@@ -39,6 +39,30 @@ const sendMessage = async ({keyPairDoc, recipientDid, message, type}) => {
     keyAgreementKey,
   });
 
+  const didDocument = await blockchainService.resolveDID(recipientDid);
+  const service = didDocument.service.find(
+    endpoint => endpoint.type === 'DIDCommMessaging',
+  );
+
+  const serviceEndpoint = service?.serviceEndpoint[0];
```

Comment:
you should loop all service endpoints until one of them is successful and has the "accept" type that you are sending with, i.e `didcomm/v2`.

Rank candidates (typical policy)
- Prefer entries advertising DIDComm v2 and HTTPS
- Respect the endpoint’s own accept order (sender should pick the most preferred media type it can satisfy).
- If there are multiple DIDCommMessaging services, prefer them in the order they appear / are advertised (common practice for “preference”).

Reply:
also you cant assume serviceEndpoint is an array, see https://identity.foundation/didcomm-messaging/spec/#service-endpoint

it could be an object too (typical for jsonld)